pkg/cli/job/list.go

```diff
@@ -104,11 +106,33 @@ func ListJobs(ctx context.Context) error {
 		return err
 	}
 
-	if len(jobs.Items) == 0 {
+	// define the filter callback function based on different flags
+	filterFunc := func(job v1alpha1.Job) bool {
+		// filter by QueueName if specified
+		if listJobFlags.QueueName != "" && listJobFlags.QueueName != job.Spec.Queue {
+			return false
+		}
+		// filter by SchedulerName if specified
+		if listJobFlags.SchedulerName != "" && listJobFlags.SchedulerName != job.Spec.SchedulerName {
+			return false
+		}
+		// filter by selector if specified
+		if listJobFlags.selector != "" && !strings.Contains(job.Name, listJobFlags.selector) {
+			return false
+		}
+		// filter by namespace if specified
```

Comment:
List has already specified the namespce, do we still need filter it?

Reply:
fixed